### ISSUE TITLE
又出现登录失败了

### ISSUE THREAD

**Issue by Gahon1995** [pjialin/py12306#155]
提示网络错误，看了一下返回的网页，内容提示的就是网络可能有问题，但是浏览器里边是能够登录的

**Comment by x1zbin** [comment]
我也是遇到这种情况了，使用浏览器能够登录，网络也没有问题；直接用docker跑的，好像是更新到了v1.0.0后就这样了

**Comment by pjialin** [comment]
最近浏览器标识生成算法一直变导致的，这两天没跟着改了，先观察观察再看。

在使用的朋友暂时可以通过手动修改代码解决

在浏览器上通过检查工具拿到 Cookie:  
`RAIL_EXPIRATION, RAIL_EXPIRATION `

覆盖代码这个地方: 
https://github.com/pjialin/py12306/blob/8d4ea3066abd1767b7ec5b3046743221156f55dc/py12306/user/job.py#L192

**Comment by Gahon1995** [comment]
好像现在官网的登陆接口不是代码中的这个了，同时好像多了一个static请求

**Comment by ZHANGBADAO** [comment]
RAIL_DEVICEID 浏览器里找见这个cookie, 复制这个cookie的值 覆盖193行的值 就可以登录了

**Comment by SparklesCN** [comment]
有个问题想问下，就是拿到cookie content之后，覆盖的是'RAIL_EXPIRATION': result.get(' '), 括号内内容吗？ 还是说要覆盖掉整个result.get。。。。

**Comment by op69qs** [comment]
这边使用免费打码接口，也是无法登录

`打码失败: 错误原因 从免费打码获取结果失败
5 秒后重新获取验证码`

也是和上面同样的原因吗？

**Comment by Gahon1995** [comment]
打码失败是因为接口的原因，可以多试几次

**Comment by op69qs** [comment]
运行1小时都是同样的报错

**Comment by yelvlab** [comment]
RAIL_EXPIRATION这个怎么获得呢

**Comment by SparklesCN** [comment]
> RAIL_EXPIRATION这个怎么获得呢

Chrome浏览器，登录12306账户之后，点击地址栏最左边 锁 的图标， 里面有cookies，然后查找对应就可以的。

**Comment by xieende** [comment]
这个deviceID 要是能自动获取就完美了

**Comment by hivefans** [comment]
使用ruokuai的接口，还是无法登录，提示

验证码验证成功 开始登录...
登录失败 错误原因: 网络错误

**Comment by mansonami** [comment]
还是原版好用，至少错误提示很明确
当前配置：
出发站：邵阳
到达站：深圳北
乘车日期：['2019-06-18']
坐席：二等座
是否有票优先提交：False
乘车人：
刷新间隔: 随机(1-3S)
购票方式：根据候选车次购买
候选购买车次：G6153,G6184,G6173
僵尸票关小黑屋时长: 5
下单接口: 2
下单模式: 2
预售踩点时间:13:00:00 
**************************************************
开启cdn查询
本次待筛选cdn总数为1857, 筛选时间大约为5-10min
需要验证码
下载验证码...
下载验证码成功
Corrupt JPEG data: 15 extraneous bytes before marker 0xd9
WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /usr/local/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:3445: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
2019-06-14 18:01:31.964483: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
/usr/local/lib/python3.7/site-packages/keras/engine/saving.py:292: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
题目为['辣椒酱']
/usr/local/lib/python3.7/site-packages/keras/engine/saving.py:292: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
1 金字塔
2 黑板
3 耳塞
4 蜜蜂
5 辣椒酱
6 耳塞
7 辣椒酱
8 蜜蜂
验证码识别坐标为40,149,184,149
40,149,184,149
fresult {'result_message': '验证码校验成功', 'result_code': '4'}
验证码通过,开始登录..
登录成功
欢迎  登录
09:20 13:21 04:01
车次: G6153 始发车站: 邵阳 终点站: 深圳北 二等座: 有
设置乘车人数为: 1
查询到有余票，尝试提交订单
使用缓存中查找的联系人信息
您还有未处理的订单，请您到<a href="../view/train_order.html">[未完成订单]</a>进行处理!

**Comment by EverydayLearning** [comment]
老哥，又出现登录错误的问题了。
![image](https://user-images.githubusercontent.com/35058785/65095738-ffe5ab00-d9f4-11e9-9782-1b1786030631.png)

**Comment by wuzhifeibuhuifei** [comment]
> 
> 
> 老哥，又出现登录错误的问题了。
> ![image](https://user-images.githubusercontent.com/35058785/65095738-ffe5ab00-d9f4-11e9-9782-1b1786030631.png)

我也是刚copy下来，然后启动出现这个问题，我在想要是一直这样尝试下载验证码会被封号吗

**Comment by Alion548** [comment]
遇到相同情况
```
正在下载验证码...
打码失败: 错误原因 从免费打码获取结果失败
5 秒后重新获取验证码
```

**Comment by liyunty** [comment]
打码的接口有问题了吧

**Comment by 13551527194** [comment]
我这边尝试也是出现了很多问题，不过建议熟悉这个项目的童鞋，可以将12306的使用chromeDriver模拟登录挪过来，我刚尝试了12306的chromeDriver，很稳（来自萌新的呐喊）

**Comment by 13551527194** [comment]
12306的本地自动识别打码，感觉已经达到了巅峰，建议熟悉项目的童鞋，一起挪过来（来自懒癌患者的独白）

**Comment by xirenbunong** [comment]
> 有个问题想问下，就是拿到Cookie内容之后，覆盖的是“ RAIL_EXPIRATION”：result.get（''），括号内内容吗？还是说要覆盖掉整个result.get。。。。
> 
> 正确格式应该为：
> 'RAIL_EXPIRATION'：'xxxxxxxx''RAIL_DEVICEID 
> '：'xxxxxxxx' 
> 这个意思吗？
> 
> 两个方法都测试过了，依然网络错误报错

额，就是在12306官网登录后，浏览器控制台的cookie的这两个字段的value，覆盖xxxxxxxx就可以
'RAIL_EXPIRATION'：'你的值'
'RAIL_DEVICEID '：'你的值'

**Comment by pjialin** [comment]
已修复，暂时先关闭了

**Comment by namewhat** [comment]
没有解决哎，我刚才复制的，也是网络错误

**Comment by superchangme** [comment]
这个cookie应该只有3三天有效期 三天后需要自己替换
